### DISCUSSION TITLE
How do I query over nodes in the injected language with treesitter?

### DISCUSSION THREAD

**Discussion by mawkler** [tree-sitter/tree-sitter#2696] [Q&A] [answered]
I'm trying to select nodes inside an injected language. Here's the example text file (Markdown):

```markdown
[[foo]]
```

When I use this query (which is also the syntax tree)...

```scm
(paragraph) ; markdown
 (inline) ; markdown
  (shortcut_link) ; markdown_inline
   (link_text) @query ; markdown_inline
```

...I get the error `"Invalid node type shortcut_link/link_text"`, since they're in an injected language. How do I query over the injected `markdown_inline` nodes?

**Comment by wren** [comment]
@mawkler Did you find an answer to this? I've been searching for this today, and am having trouble finding an answer.

Here's a similar question I found with no answer: https://neovim.discourse.group/t/treesitter-query-that-matches-nodes-in-embedded-language-vue/4395

**Reply by mawkler** [reply]
Not really, as far as I can remember

**Reply by wren** [reply]
> Not really, as far as I can remember

Thanks for the quick reply! I'm sorry to hear there was no solution found. I'll keep looking, and update here if I find anything.

**Reply by amaanq** [reply]
It sounds like this question should be raised in a Neovim discussion/channel.

**Reply by wren** [reply]
@amaanq The Neovim thread linked above is just one example of the question. The general question definitely applies to treesitter as a whole, though, so I don't think this question should be closed, yet.

The question is this: 
Is there an intended way to query a syntax tree that contains an injected language?

**Reply by amaanq** [reply, marked as answer]
In general, injections aren't really handled upstream, they are largely an implementation detail of the user -- we only use injections for the `tree-sitter-highlight` crate, which doesn't expose the parse trees to the user, so there hasn't been a need or answer for this question. It's really up to the implementation for how to handle querying the injected syntax tree, but ideally, you should be able to do so.
